accessible/fieldview.component.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * AccessibleBlockly
+ *
+ * Copyright 2016 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Angular2 Component that details how a Blockly.Field is
+ * rendered in the toolbox in AccessibleBlockly. Also handles any interactions
+ * with the field.
+ * @author madeeha@google.com (Madeeha Ghori)
+ */
+blocklyApp.FieldView = ng.core
+  .Component({
+    selector: 'field-view',
+    template: `
+    <li #listItem aria-selected=false role="treeitem" [attr.aria-level]="level" *ngIf="isTextInput(field)" id="{{treeService.createId(listItem)}}">
+      <input #input id="{{treeService.createId(input)}}" [ngModel]="field.getValue()" (ngModelChange)="field.setValue($event)">
+      {{setLabelledBy(listItem, concatStringWithSpaces("blockly-argument-input", input.id))}}
+    </li>
+    <li #listItem aria-selected=false role="treeitem" [attr.aria-level]="level" *ngIf="isDropdown(field)" id="{{treeService.createId(listItem)}}">
+      <label #label id="{{treeService.createId(label)}}">current argument value: {{field.getText()}}</label>
+      <ol role="group"  [attr.aria-level]="level+1">
+        <li #option *ngFor="#optionValue of getOptions(field)" id="{{treeService.createId(option)}}" role="treeitem" aria-selected=false [attr.aria-level]="level+1">
+          <button #optionButton id="{{treeService.createId(optionButton)}}" (click)="handleDropdownChange(field,optionValue)">{{optionText[optionValue]}} button</button>
+        </li>
+      </ol>
+      {{setLabelledBy(listItem, concatStringWithSpaces("blockly-argument-menu", label.id))}}
```

Comment:
If you'll notice, the variable `input` is being defined on line 31 and used on line 32. In order to set aria-labelledby more directly, I would have to define `input` on line 31 and use it on line 30. So I devised this system to get around that, since I have gotten errors along similar issues in the past. That said, I just tried setting aria-labelledby directly and for some reason it works even if the variable is being defined a line after it's being used. I will make those changes in the next CL.